casper/src/main/scala/coop/rchain/casper/BlockCreator.scala

```diff
@@ -73,7 +72,13 @@ object BlockCreator {
         now              <- Time[F].currentMillis
         invalidBlocksSet <- dag.invalidBlocks
         invalidBlocks    = invalidBlocksSet.map(block => (block.blockHash, block.sender)).toMap
-        unsignedBlock <- if (deploys.nonEmpty || slashingDeploys.nonEmpty) {
+        // make sure closeBlock is the last system Deploy
+        systemDeploys = if (slashingDeploys.nonEmpty)
+          slashingDeploys ++ Seq(
+            CloseBlockDeploy(Tools.rng(parents.head.blockHash.toByteArray))
```

Comment:
@zsluedemI if several blocks have the same main parent (e.g. the very first blocks that all have genesis as a parent) - CloseBlockDeploy deploy will be initialised with the same seed. Isn't this a problem?

Reply:
I actually already had a fix for that. But Rao said it should be cautious about it. So I didn't bring up.